clippy_lints/src/methods/mod.rs

```diff
@@ -4230,7 +4257,12 @@ impl Methods {
                 ("as_mut", []) => useless_asref::check(cx, expr, "as_mut", recv),
                 ("as_ref", []) => useless_asref::check(cx, expr, "as_ref", recv),
                 ("assume_init", []) => uninit_assumed_init::check(cx, expr, recv),
-                ("cloned", []) => cloned_instead_of_copied::check(cx, expr, recv, span, &self.msrv),
+                ("cloned", []) => {
+                    cloned_instead_of_copied::check(cx, expr, recv, span, &self.msrv);
+                    if let Some((name @ ("as_ref" | "as_mut"), recv, .., as_ref_ident_span)) = method_call(recv) {
```

Comment:
nit: I'd personally put this inside `option_as_ref_cloned::check`, but it's fine either way

Reply:
It's similar in a few other places like here
https://github.com/rust-lang/rust-clippy/blob/e1dbafd875f898a14514b39c928e735fb5a7e6b4/clippy_lints/src/methods/mod.rs#L4392-L4393

but I can also move it